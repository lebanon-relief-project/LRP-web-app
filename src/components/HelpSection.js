import React from "react";
import styled from "styled-components";
import Card from "./Card";

const HelpSection = () => {
  return (
    <StyledSection>
      <CardWrapper>
        <Card
          title="I’m looking for help"
          text="Lorem lobortis aliquam leo nisi vitae egestas elit. Tellus ultrices et velit luctus. "
        />
        <Card
          title="I’d like to help"
          text="Lorem lobortis aliquam leo nisi vitae egestas elit. Tellus ultrices et velit luctus. "
        />
      </CardWrapper>
    </StyledSection>
  );
};

const CardWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
`;

const StyledSection = styled.section`
  background: #fafafa;
  margin: 0;
  padding: 0;
  min-height: 10%;
`;

export default HelpSection;
