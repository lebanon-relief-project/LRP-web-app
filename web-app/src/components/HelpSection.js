import React, { useState, useRef, useEffect } from "react";
import styled from "styled-components";
import Card from "./Card";
import LikeToHelpModal from "./LikeToHelpModal";
import { Routes } from "../App.js";
import devices from "../styles/Devices";
import { Link } from "react-router-dom";
import colours from "../styles/Colours";

const HelpSection = () => {
  const [height, setHeight] = useState(0);
  const ref1 = useRef(null);
  const ref2 = useRef(null);
  const ref3 = useRef(null);

  let data = [
    {
      title: "I’m looking for help",
      content: [
        "Everyone reacts differently to trauma, and not everyone recovers from a traumatic experience in the same way or time. There is no one solution to heal from it.",
        "Here you can find personalized recommendations, tips and tools to help you cope with trauma.",
      ],
    },
  ];

  useEffect(() => {
    let maxHeight = Math.max([
      ref1.current.clientHeight,
      ref2.current.clientHeight,
      ref3.current.clientHeight,
    ]);
    if (!isNaN(maxHeight)) setHeight(maxHeight);
    else setHeight(ref3.current.clientHeight);
  }, []);

  return (
    <>
      <StyledSection>
        <div
          style={{
            maxWidth: 960,
            display: "flex",
            backgroundColor: "inherit",
            gap: 24,
            flexWrap: "wrap",
            justifyContent: "center",
            paddingTop: 16,
          }}
        >
          <CardWrapper
            ref={ref1}
            style={{
              height: height > 0 ? height : "auto",
            }}
          >
            <TitleWrapper>I’m looking for help</TitleWrapper>
            <div style={{ marginTop: 10 }}>
              <div>
                Everyone reacts differently to trauma, and not everyone recovers
                from a traumatic experience in the same way or time. There is no
                one solution to heal from it.
              </div>
              <div style={{ marginTop: 16 }}>
                Here you can find personalized recommendations, tips and tools
                to help you cope with trauma.
              </div>
            </div>

            <LinkWrapper>
              <StyledLink to={"/"}>Give me advice</StyledLink>
            </LinkWrapper>
          </CardWrapper>

          <CardWrapper
            ref={ref2}
            style={{
              height: height > 0 ? height : "auto",
            }}
          >
            <TitleWrapper>I’d like to help</TitleWrapper>
            <div style={{ marginTop: 10 }}>
              <div>
                Do you have a close one who underwent trauma? Do you need
                direction on how to help, what to say, what to do, and how to
                protect yourself?
              </div>
              <div style={{ marginTop: 16 }}>
                Or do you want to help those who are dealing with trauma, and
                are looking for opportunities to do so?
              </div>
            </div>

            <LinkWrapper>
              <StyledLink to={"/"}>Contact me</StyledLink>
            </LinkWrapper>
          </CardWrapper>

          <CardWrapper
            ref={ref3}
            style={{
              height: height > 0 ? height : "auto",
            }}
          >
            <TitleWrapper
              style={{
                width: "60%",
              }}
            >
              I want to talk to someone
            </TitleWrapper>
            <div style={{ marginTop: 38 }}>
              <div>
                Sometimes, self-help is not enough, and you might prefer a
                professional hand to help you cope with trauma.
              </div>
              <div style={{ marginTop: 16 }}>
                We created a directory of mental health experts to help you find
                and connect to the therapist, counselling service, or
                organisation that resonates best with you.
              </div>
            </div>

            <LinkWrapper>
              <StyledLink to={"/"}>View directory</StyledLink>
            </LinkWrapper>
          </CardWrapper>
        </div>
      </StyledSection>
    </>
  );
};

const StyledSection = styled.div`
  display: flex;
  justify-content: center;
  background: inherit;
  margin: 0;
  width: 100vw;
  margin-left: 50%;
  transform: translateX(-50%);
`;

const TitleWrapper = styled.div`
  background-color: inherit;
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: -16px;
  width: auto;
  align-self: center;
  font-size: 24px;
  font-family: Playfair Display;
  font-weight: 700;
  height: auto;
  line-height: 32px;
  text-align: center;
  padding-left: 10px;
  padding-right: 10px;
`;

const CardWrapper = styled.div`
  background-color: inherit;
  width: 304px;
  padding: 20px;
  display: flex;
  flex-direction: column;
  font-family: "Raleway";
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  border: 1px solid #003a8c;
  position: relative;
`;

const LinkWrapper = styled.div`
  display: flex;

  flex: 1;
  align-items: flex-end;
`;

const StyledLink = styled(Link)`
  background-color: ${colours.yellow};
  color: black;
  font-style: normal;
  font-weight: bold;
  justify-content: center;
  align-items: center;
  display: flex;
  height: 40px;
  width: 100%;
  margin-top: 24px;
  justify-self: flex-end;
`;

export default HelpSection;
