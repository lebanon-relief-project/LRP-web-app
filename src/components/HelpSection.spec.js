import React from "react";
import { render } from "@testing-library/react";
import HelpSection from "./HelpSection";

describe("the Card component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderHelpSection());

    expect(container).toMatchSnapshot();
  });
});

const renderHelpSection = () => {
  return render(<HelpSection />);
};
