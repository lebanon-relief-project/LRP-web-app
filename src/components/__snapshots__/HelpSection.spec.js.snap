// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the HelpSection component should match the snapshot 1`] = `
.c3 {
  font-weight: bold;
}

.c2 {
  border: 1px solid grey;
  width: 25%;
  padding: 1rem;
  margin: 3rem 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c0 {
  background: #fafafa;
  margin: 0;
  padding: 0;
  min-height: 10%;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          I’m looking for help
        </h2>
        <p>
          Lorem lobortis aliquam leo nisi vitae egestas elit. Tellus ultrices et velit luctus. 
        </p>
        <button>
          Learn More
        </button>
      </div>
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          I’d like to help
        </h2>
        <p>
          Lorem lobortis aliquam leo nisi vitae egestas elit. Tellus ultrices et velit luctus. 
        </p>
        <button>
          Learn More
        </button>
      </div>
    </div>
  </section>
</div>
`;
